### DISCUSSION TITLE
thermostat with multiple am2301 and relais

### DISCUSSION THREAD

**Discussion by bsinfo523** [arendst/Tasmota#16767] [Q&A]
Hi,

I flashed an esp32 with tasmota thermostat firmware.
Program Version 12.1.1(tasmota)

I connected an 8 channel relay board to the esp32 and also two temperature sensors (AM2301) for testing.

I have already figured out how to activate the individual thermostats. (found here in another discussion :))

cmnd/Tasmota_Name/THERMOSTATMODESET1 1
cmnd/Tasmota_Name/THERMOSTATMODESET2 1
cmnd/Tasmota_Name/THERMOSTATMODESET3 1

cmnd/Tasmota_Name/TEMPTARGETSET1 22.5
cmnd/Tasmota_Name/TEMPTARGETSET2 25
cmnd/Tasmota_Name/TEMPTARGETSET3 20

cmnd/Tasmota_Name/SENSORINPUTSET1 1 #1=local 0=mqtt
cmnd/Tasmota_Name/SENSORINPUTSET2 1
cmnd/Tasmota_Name/SENSORINPUTSET3 1

1. Question
How can I now set which temperature sensor is assigned to which thermostat?

2. Question
How can i change the name of the thermostats? All Thermostats are called "Thermostat"?!

3. Question
How can i assign a schedule with a temp slider(to set the TEMPTARGETSET) for each Thermostat?

![tasmota_multiple_thermostatsenaled](https://user-images.githubusercontent.com/9340874/194745744-11c28db6-69bd-415f-94a1-9a6a527a8e25.jpg)

**Comment by barbudor** [comment]
> Question 1 : How can I now set which temperature sensor is assigned to which thermostat?

Don't use `SensorInputSet<x> 1` which works only for 1 local sensor. Not applicable to your use-case as you have multiple sensors. Stick with `SensorInputSet<x> 0` default mode.
In this default mode Thermostat is expected to receive the temperature through the command `TempMeasuredSet`. Don't let you fool by the doc, it is not mandatory to receive this command over MQTT. It also works locally from a rule.
So you should be able to use a rule such as (theory, not tested):
```
Rule1 on tele-AM2103-22#Temperature do TempMeasuredSet1 %value% endon on tele-AM2103-23#Temperature do TempMeasuredSet2 %value% endon
Rule1 1
```
Note1: for more sensors & thermostats, adapt the rules. I think you get the idea.
Note2: I'm using the `tele-` prefix to have the rule executed only at Teleperiod which I believe is enough to let the Thermostat aware of temperature changes. Adapt Teleperiod to your use case.


> Question 2 : How can i change the name of the thermostats? All Thermostats are called "Thermostat"?!

I don't see any option for that .
That could be the purpose for a new feature request which I don't see too complicated to implement.

> Question 3 : How can i assign a schedule with a temp slider(to set the TEMPTARGETSET) for each Thermostat?

Tasmota doesn't have a complex GUI for Thermostat.
You can use [Timers](https://tasmota.github.io/docs/Timers/) which can be configured from the WebUI in Configure => Configure Timer
Is you select the Action for a Timer to a Rule (instead of ON/OFF) you can then have a rule triggered for changing the TempTargetSet for example:
```
Rule2 on clock#timer=1 do TempTargetSet1 25.4 endon
```
You can associate that with variables ([var or mem](https://tasmota.github.io/docs/Rules/#rule-variables)) to represent typical values such as:
mem1 = Bedroom Eco temp
mem2 = Normal Temp
mem3 = Living Room Eco temp
mem4 = Living Room Temp
And use that as `%mem1%` in the rule instead of hardcoded values, easier to change from Console or MQTT than changing the rule.

To go further on the GUI you have 2 options:
- Use your favorite home automation software to have a true GUI with sliders etc...
- Use [Berry](https://tasmota.github.io/docs/Berry/) language to define your own Web GUI extensions 
  In the Berry [exemples](https://github.com/arendst/Tasmota/tree/development/tasmota/berry/examples) there 2 examples regarding that GUI extension topic
